src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1901,13 +1901,17 @@ private function addHttpClientSection(ArrayNodeDefinition $rootNode, callable $e
                                     ->ifTrue(function ($v) { return !empty($v['query']) && !isset($v['base_uri']); })
                                     ->thenInvalid('"query" applies to "base_uri" but no base URI is defined.')
                                 ->end()
+                                ->validate()
+                                    ->ifTrue(fn ($v) => (isset($v['base_uri']) && \is_array($v['base_uri'])) && !isset($v['retry_failed']))
+                                    ->thenInvalid('"base_uri" can only be an array if "retry_failed" is defined.')
```

Comment:
defined or enabled for retry_failed?
also, should we validate the array contains only strings?

Reply:
I'd say that more precautions are always better, so indeed checking if there are strings only!
And yeah, retries should be enabled, imo.

Going to make changes in that direction.

Reply:
In this code we're merging all validations but the error message is the same for all. If you define an array and one of the values is no a string by mistake, you'll see this and will be confusing:

```
"base_uri" can only be an array if "retry_failed" is defined
```

Maybe we could extract this validation and create a dedicated error message for it?

```php
\count($v['base_uri']) !== \count(array_filter($v['base_uri'], 'is_string'))
```

Thanks!